### PR TITLE
RTMP audio is only visible in the microphone settings when RTMP is connected

### PR DESCRIPTION
**Pull request by ErAzOr2k**
RTMP audio is only visible in the microphone settings when RTMP is connected.
When RTMP is disconnected, the previously selected microphone is selected

### CODE REVIEW

File: Moblin/Various/Model.swift [hunk 1]
```diff
@@ -1197,20 +1198,23 @@ final class Model: NSObject, ObservableObject {
             if self.database.debug!.enableRtmpAudio! {
                 self.media.addRtmpAudio(cameraId: stream.id)
             }
+            stream.isConnected = true
```
Comment (eerimoq) [inline Moblin/Various/Model.swift:1201]: Hmm, is it really necesary to keep track of the RTMP stream status in the stream "settings". I mean, isn't it possible to call `isRtmpStreamConnected()` on the model somehow?

I have not looked into the details though, but prefer keeping the connection state in one place.

Comment (ErAzOr2k) [reply Moblin/Various/Model.swift:1201]: cool, I didn't discover the function. That is of course much better. will adjust it accordingly.

Comment (eerimoq) [reply Moblin/Various/Model.swift:1201]: I hope it works as I didn't look too closely at it. Too much in Moblin, so hard to remember everything =)

File: Moblin/Various/Model.swift [hunk 2]
```diff
@@ -1204,13 +1205,14 @@ final class Model: NSObject, ObservableObject {
...
+            guard let stream = self.getRtmpStream(streamKey: streamKey) else {
                 return
             }
-            self.media.removeRtmpCamera(cameraId: cameraId)
+            self.media.removeRtmpCamera(cameraId: stream.id)
             if self.database.debug!.enableRtmpAudio! {
-                self.media.removeRtmpAudio(cameraId: cameraId)
+                self.media.removeRtmpAudio(cameraId: stream.id)
             }
+            self.selectMic(mic: self.previousMic)
```
Comment (eerimoq) [inline Moblin/Various/Model.swift:1215]: What happens if the previous mic also was an RTMP mic, and that mic can no longer be used as that RTMP stream was disconnected some time ago?

Comment (eerimoq) [reply Moblin/Various/Model.swift:1215]: Or if it was an bluetooth mic. The same logic can be applied to those I think.

Comment (eerimoq) [reply Moblin/Various/Model.swift:1215]: It might be easiest to just start using a builtin mic if an external or RTMP mic is disconnected.

Comment (ErAzOr2k) [reply Moblin/Various/Model.swift:1215]: I suggest I switch to the microphone, which is saved in the settings. that would be front, bottom or back

Comment (eerimoq) [reply Moblin/Various/Model.swift:1215]: sounds good. didn't remember that exists =)

Comment (ErAzOr2k) [reply Moblin/Various/Model.swift:1215]: or top of course :)

File: Moblin/Various/Model.swift [hunk 3]
```diff
@@ -1204,13 +1204,14 @@ final class Model: NSObject, ObservableObject {
...
+            guard let stream = self.getRtmpStream(streamKey: streamKey) else {
                 return
             }
-            self.media.removeRtmpCamera(cameraId: cameraId)
+            self.media.removeRtmpCamera(cameraId: stream.id)
             if self.database.debug!.enableRtmpAudio! {
-                self.media.removeRtmpAudio(cameraId: cameraId)
+                self.media.removeRtmpAudio(cameraId: stream.id)
             }
+            self.setMicFromSettings()
```
Comment (eerimoq) [inline Moblin/Various/Model.swift:1214]: Maybe move this into the if statement above? The case I'm thinking about is is the user want to use an external bluetooth mic with RTMP video. If that case the external mic should continue to be used after the RTMP stream stops.

Comment (ErAzOr2k) [reply Moblin/Various/Model.swift]: OK, I changed it. had to add an additional check

**Pull request merged**